File: 4.cpp
double findMedianSortedArrays(int* nums1, int nums1Size, int* nums2, int nums2Size) {
    int i = 0, j = 0, tot = 0;
    int tmp[10010];
    while(i < nums1Size&&j < nums2Size){
        if(nums1[i] <= nums2[j]){
            tmp[++tot] = nums1[i++];
        }
        else{
            tmp[++tot] = nums2[j++];
        }
    }
    while(i < nums1Size)tmp[++tot] = nums1[i++];
    while(j < nums2Size)tmp[++tot] = nums2[j++];
    if((nums1Size + nums2Size) % 2)
        return tmp[((nums1Size + nums2Size) / 2) + 1];
    else 
        return (tmp[(nums1Size + nums2Size) / 2] + tmp[((nums1Size + nums2Size) / 2) + 1]) / 2;
}

File: 4.py
#虽然用c过了但还是写一下python 还不太会用python的列表
class Solution:
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        n = len(nums1)
        m = len(nums2)
        mid = int((n + m) / 2)
        tmp = [0]
        i = 0
        j = 0
        while  i < n and j < m:
            if nums1[i] < nums2[j]:
                tmp.append(nums1[i])
                i += 1
            else :
                tmp.append(nums2[j])
                j += 1
        while i < n:
            tmp.append(nums1[i])
            i += 1
        while j < m:
            tmp.append(nums2[i])
            j += 1
        if (n + m) % 2:
            return tmp[mid + 1]
        else :
            return (tmp[mid] + tmp[mid+1]) / 2
        